.github/workflows/Dockerfile

```diff
@@ -1,17 +1,19 @@
+ARG BASE_OS
+ARG BASE_DISTRO
 ARG BASE_IMAGE
 ARG K_VERSION
 ARG Z3_VERSION
-FROM z3:${BASE_IMAGE}-${Z3_VERSION} as Z3
+FROM z3:${BASE_DISTRO}-${Z3_VERSION} as Z3
 
-ARG BASE_IMAGE
+ARG BASE_OS
 ARG K_VERSION
-FROM stack:${BASE_IMAGE}-${K_VERSION} as STACK
+FROM stack:${BASE_DISTRO}-${K_VERSION} as STACK
 
-ARG BASE_IMAGE
+ARG BASE_OS
```

Comment:
Same here, you need to redeclare `BASE_DISTRO` variable again.

Reply:
Strange Nuance I have not run into before.